zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/client/ZkClient.java

```diff
@@ -90,7 +90,15 @@ public ZkClient(IZkConnection zkConnection, int connectionTimeout, long operatio
       PathBasedZkSerializer zkSerializer,
       String monitorType, String monitorKey, String monitorInstanceName,
       boolean monitorRootPathOnly) {
-    super(zkConnection, connectionTimeout, operationRetryTimeout, zkSerializer, monitorType,
+    super(null, zkConnection, connectionTimeout, operationRetryTimeout, zkSerializer, monitorType,
+        monitorKey, monitorInstanceName, monitorRootPathOnly);
+  }
+
+  public ZkClient(String realmKey, IZkConnection zkConnection, int connectionTimeout, long operationRetryTimeout,
```

Comment:
I don't think making the raw ZkClient realm-aware is appropriate because it's supposed to be a "raw" ZkClient that preserves the behavior of what we had in I0Itec's ZkClient.